source/fe/mapping_fe_field.cc

```diff
@@ -202,6 +202,22 @@ MappingFEField<dim,spacedim,DoFHandlerType,VectorType>::InternalData::fourth_der
 
 
 
+namespace
+{
+  // Construct a quadrature formula containing the vertices of the reference
+  // cell in dimension dim (with invalid weights)
+  template<int dim>
+  Quadrature<dim> get_vertex_quadrature()
+  {
+    std::vector<Point<dim> > points(GeometryInfo<dim>::vertices_per_cell);
+    for (unsigned int i=0; i<GeometryInfo<dim>::vertices_per_cell; ++i)
+      points[i] = GeometryInfo<dim>::unit_cell_vertex(i);
+    return Quadrature<dim>(points);
```

Comment:
Isn't this what `QTrapez` already does?

I'm wondering about this because the code you have here creates the same quadrature object every time. This might not matter because we don't create `MappingFEField` very often, but it's still annoying. It would be nice if you could just write something like
```
template <int dim>
const Quadrature<dim> &     // note: reference!
get_vertex_quadrature ()
{
  static const QTrapez<dim> q;
  return q;
}
```
which would create the quadrature formula only once.

Reply:
QTrapez does not assure that the *order* of the quadrature points is the same as the order of the vertices, or does it? I'll make the code return a reference to a static object.